src/Festung/Vault/VaultHandler.hs

```diff
@@ -109,6 +113,26 @@ query o v q = flattenError <$> sendMappedCommand f v q
 
 
 -- | Execute a parametrized query
-parametrizedQuery :: VaultOpener -> Vault -> String -> [P.Value] -> IO (Either VaultError P.QueryResult)
-parametrizedQuery o v q p = flattenError <$> sendMappedCommand f v (q, p)
+parametrizedQuery :: VaultOpener -> Vault -> String -> P.QueryParams
+                  -> IO (Either VaultError P.QueryResult)
+parametrizedQuery o v q (P.Params p) = flattenError <$> sendMappedCommand f v (q, p)
     where f = (,) o . ParametrizedQuery
+parametrizedQuery o v q (P.BulkParams bps) = 
+    _parametrizedQuery o v q (P.BulkParams bps) (Right P.emptyQueryResult)
+
+
+_parametrizedQuery :: VaultOpener -> Vault -> String -> P.QueryParams
+                   -> Either VaultError P.QueryResult -> IO (Either VaultError P.QueryResult)
+_parametrizedQuery o v q (P.BulkParams []      ) res    = return res
```

Comment:
I dont fully understand the usage of whitespace here. I would more opt for something like this
```haskell
_parametrizedQuery o v q (P.BulkParams [])     currRes = return currRes
_parametrizedQuery o v q (P.BulkParams (p:bp)) prevRes = do
```
(I secretly renamed `res` to `currRes`)